test/SetAPI_test.py

```diff
@@ -0,0 +1,147 @@
+"""General save and get tests of the SetAPI.
+
+See save_set_error_test.py and get_set_error_test.py for error handling tests.
+"""
+from test.common_test import check_get_set, check_get_set_output, check_save_set_output
+from test.conftest import SET_FIXTURE_MAP
+from typing import Any
+
+import pytest
+from _pytest.fixtures import FixtureRequest
+from SetAPI.generic.constants import INC_ITEM_INFO, INC_ITEM_REF_PATHS, REF_PATH_TO_SET
+from SetAPI.SetAPIImpl import SetAPI
+
+
+@pytest.mark.parametrize(
+    ("ws_id", "data_fixture"),
+    [("default_ws_id", data_fixture) for data_fixture in SET_FIXTURE_MAP],
+    indirect=["ws_id", "data_fixture"],
+)
+def test_save_set(ws_id: int, data_fixture: dict[str, Any]) -> None:
+    """Test that saving a set produces the expected output.
+
+    This saves sets in the default workspace (default_ws_id) and uses the
+    fixtures corresponding to the keys in the SET_FIXTURE_MAP mapping.
+    See test/conftest.py for SET_FIXTURE_MAP.
+
+    :param ws_id: workspace ID of the workspace that the sets are saved in
+    :type ws_id: int
+    :param data_fixture: name of the data fixture to use in the test
+    :type data_fixture: dict[str, Any]
+    """
+    check_save_set_output(**data_fixture)
+
+
+@pytest.mark.parametrize("ws_id", ["default_ws_id"], indirect=["ws_id"])
+@pytest.mark.parametrize("set_name", list(SET_FIXTURE_MAP))
+def test_get_set_all_set_types(
```

Comment:
this is a new test - runs through all the different set types and makes sure they can all be fetched.